flexmeasures/data/models/reporting/pandas_reporter.py

```diff
@@ -87,8 +85,17 @@ def _compute(
             ]
             final_output = tb.BeliefsDataFrame(timed_beliefs)
 
-        if self.output_event_resolution:
-            final_output.event_resolution = self.output_event_resolution
+        # use event_starts frequency when final_output event resolution
+        # does not correspond with the event resolution of the output sensor
+        event_frequency = final_output.event_starts.inferred_freq
+
+        if event_frequency:
+            event_frequency = pd.to_timedelta(
+                pd.tseries.frequencies.to_offset(event_frequency)
+            ).to_pytimedelta()
```

Comment:
I think this could be problematic. The [suggested refactoring of the reporter class](https://github.com/FlexMeasures/flexmeasures/discussions/662#discussioncomment-5803785) may shine some light, but there are some other things we also need to consider:
- [The distinction between frequency and resolution](https://flexmeasures.readthedocs.io/en/latest/api/notation.html#frequency-and-resolution), at least in the `TimelyBeliefsReporter`.
- FlexMeasures stores resolutions in the database as [intervals](https://www.postgresql.org/docs/current/functions-datetime.html), which allows distinguishing between a calendar day (`interval '1 day'`) and a 24-hour day (`interval '24 hours'`), but uses mostly Python `datetime.timedelta` objects in the code, which doesn't allow expressing a calendar day (`timedelta(days=1)` is defined to be exactly `timedelta(hours=24)`). So Pandas offsets won't always end up correctly in our database. And even within Pandas, some assumptions regarding these conversions can be challenged. [For example, in the case of "24H" = timedelta(hours=24).](https://github.com/pandas-dev/pandas/issues/35248)

We could use some test cases to see what works and what doesn't. And maybe we should prioritize the refactoring.